pkg/serviceprovider/github/github.go

```diff
@@ -122,6 +131,105 @@ func (g *Github) GetServiceProviderUrlForRepo(repoUrl string) (string, error) {
 	return serviceprovider.GetHostWithScheme(repoUrl)
 }
 
+func (g *Github) CheckRepositoryAccess(ctx context.Context, cl client.Client, accessCheck *api.SPIAccessCheck) *api.SPIAccessCheckStatus {
+	repoUrl := accessCheck.Spec.RepoUrl
+	publicRepo := g.publicRepo(ctx, accessCheck)
+	status := &api.SPIAccessCheckStatus{
+		RepoURL:         repoUrl,
+		Accessible:      publicRepo,
+		Private:         !publicRepo,
+		Type:            api.SPIRepoTypeGit,
+		ServiceProvider: api.ServiceProviderTypeGitHub,
+	}
+
+	lg := log.FromContext(ctx)
+
+	tokens, lookupErr := g.lookup.Lookup(ctx, cl, accessCheck)
+	if lookupErr != nil {
+		lg.Error(lookupErr, "failed to lookup token for accesscheck", "accessCheck", accessCheck)
+		return status
+	}
+
+	if len(tokens) > 0 {
+		token := &tokens[0]
+		ghClient, err := g.createAuthenticatedGhClient(ctx, token)
+		if err != nil {
+			status.ErrorReason = api.SPIAccessCheckErrorUnknownError
+			status.ErrorMessage = err.Error()
+			return status
+		}
+		owner, repo, err := g.parseGithubRepoUrl(accessCheck.Spec.RepoUrl)
+		if err != nil {
+			status.ErrorReason = api.SPIAccessCheckErrorBadURL
+			status.ErrorMessage = err.Error()
+			return status
+		}
+
+		ghRepository, _, err := ghClient.Repositories.Get(ctx, owner, repo)
+		if err != nil {
+			status.ErrorReason = api.SPIAccessCheckErrorRepoNotFound
+			status.ErrorMessage = err.Error()
+			return status
+		}
+
+		status.Accessible = true
+		status.Private = *ghRepository.Private
+		status.Tokens = make([]string, 0)
+		for _, t := range tokens {
+			status.Tokens = append(status.Tokens, t.Name)
```

Comment:
I'm not sure if this information is useful to anyone, because the outside users don't have a say in what token we pick for a certain binding... `Accessible` says it all from the user's standpoint imho.

Reply:
make sense, removed